app/src/main/java/org/glucosio/android/presenter/ExportPresenter.java

```diff
@@ -56,84 +56,78 @@
     private int toDay;
     private int toMonth;
     private int toYear;
-    private DatabaseHandler dB;
-    private MainActivity activity;
-
-    public ExportPresenter(MainActivity exportActivity, DatabaseHandler dbHandler) {
-        this.activity = exportActivity;
-        dB = dbHandler;
-    }
 
-    public void onExportClicked(final Boolean all) {
-        final List<GlucoseReading> readings;
+    private Activity mActivity;
+    private ExportView mExportView;
+    private DatabaseHandler dB;
 
-        if (all) {
-            readings = dB.getGlucoseReadings();
+    public ExportPresenter(Activity activity) {
```

Comment:
Yes it would be easy to abstract the Android apis from the Presenter if there was no Permission handling there (and it shouldn't).
With the current setup you'll be instantiating something like this:
`ExportPresenter(this, this);` not pretty, but refactoring this wouldn't make the PR with to many differences.